lib/SPIRV/libSPIRV/SPIRVDecorate.h

```diff
@@ -205,6 +205,34 @@ class SPIRVDecorate : public SPIRVDecorateGeneric {
   }
 };
 
+class SPIRVDecorateId : public SPIRVDecorateGeneric {
```

Comment:
`SPIRVDecorateId` and `SPIRVDecorate` are kind of the same. Instead of creation of another class, we can also change `SPIRVDecorate` to be templated. I didn't implement it this way tho, because addition of a template will create a bit of a mess in the code below and IMO grepping for "SPIRVDecorateId" will make implementation of other extensions faster and easier than grepping for templated "SPIRVDecorate".